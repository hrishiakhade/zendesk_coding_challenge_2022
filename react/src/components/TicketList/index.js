import React from 'react'
import { formatDate } from '../../utils/common-helpers';

const TicketList = (props) => (
    <div class="list-group">
        {props?.currentPageTickets?.map(item => {
            return (
                <a key={item.id} className="list-group-item list-group-item-action " aria-current="true" onClick={()=>props.onClickTicket(item)} >
                    <div class="d-flex w-100 justify-content-between">
                        <small>ID#{item.id}</small>
                        <h5 class="mb-1">{item.subject}</h5>
                        <small>{formatDate(item.created_at)}</small>
                    </div>
                    <div class="d-flex w-100 justify-content-between">
                        <p class="mb-1" className="ticket-item-description">{item.description}</p>
                        <small>Status:{item.status.charAt(0).toUpperCase() + item.status.slice(1)}</small>
                    </div>
                </a>
            )
        })}
    </div>
)

export default TicketList