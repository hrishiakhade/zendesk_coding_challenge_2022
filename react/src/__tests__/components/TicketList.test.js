import React from 'react';
import Enzyme, { shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJSON from 'enzyme-to-json'
import TicketList from '../../components/TicketList';
import ticketData from '../../utils/fixtures/ticketData'

Enzyme.configure({
  adapter: new Adapter()
})


test('Should render TicketList title correctly', () => {
  const wrapper = shallow(<TicketList currentPageTickets={[ticketData]}/>)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

