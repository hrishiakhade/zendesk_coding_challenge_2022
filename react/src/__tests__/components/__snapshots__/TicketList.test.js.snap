// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render TicketList title correctly 1`] = `
<div
  class="list-group"
>
  <a
    aria-current="true"
    className="list-group-item list-group-item-action "
    key="95"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      class="d-flex w-100 text-left align-items-center pt-1 pb-1"
    >
      <span
        className="w-10 id-style"
      >
        ID#
        95
      </span>
      <div
        className="w-80"
      >
        <h5
          className="mb-1 ticket-title-description"
        >
          ex cillum cillum proident commodo
        </h5>
        <p
          className="ticket-item-description mb-0"
        >
          Culpa amet mollit esse tempor sit ullamco sit ex dolore nostrud consectetur. Amet ad aliquip aliqua aute quis sunt laborum excepteur occaecat. Enim cillum do reprehenderit cupidatat ex eu ea ipsum ullamco consequat dolore duis consectetur. Magna sit voluptate laboris nisi do.

Dolor voluptate anim consequat ad cillum excepteur nisi elit. Ea tempor ipsum sint fugiat dolor anim amet pariatur aliqua labore ipsum. Sunt cillum eu fugiat est cillum mollit ullamco dolor anim in cupidatat cupidatat. Commodo eiusmod est et dolore enim do Lorem sunt quis magna sunt excepteur. Aliqua esse velit nulla labore culpa reprehenderit magna cupidatat aliquip. Do aliquip elit adipisicing aute excepteur culpa. Ullamco elit enim duis ut Lorem proident qui cupidatat magna.
        </p>
      </div>
      <div
        className="w-10"
      >
        <span
          className="d-flex mb-1 date"
        >
           Nov 21 2021
        </span>
        <small
          className="status-pill open"
        >
          Open
        </small>
      </div>
    </div>
  </a>
</div>
`;
